data/plugins/autocomplete.lua

```diff
@@ -23,6 +23,60 @@ config.plugins.autocomplete = common.merge({
   desc_font_size = 12
 }, config.plugins.autocomplete)
 
+local settings_loaded, settings = pcall(require, "plugins.settings")
```

Comment:
What if instead of having to do the `pcall` here, the settings layout is provided inside the plugin config itself?
Something like
```lua
local config_layout = {
  name = "Autocomplete",
  {
     label = "Minimum Length",
     description = "Amount of characters that need to be written for autocomplete to popup.",
     path = "min_len",
     type = settings.type.NUMBER,
     default = 3,
     min = 1,
     max = 5
  },
  ...
}

config.plugins.autocomplete = common.merge({
  -- Amount of characters that need to be written for autocomplete
  min_len = 3,
  -- The max amount of visible items
  max_height = 6,
  -- The max amount of scrollable items
  max_suggestions = 100,
  -- Maximum amount of symbols to cache per document
  max_symbols = 4000,
  -- Font size of the description box
  desc_font_size = 12,
  config_layout = config_layout,
}, config.plugins.autocomplete)
```
Then the settings plugin can just scan the `config.plugins` table to obtain the layout.

Reply:
that is actually a good idea, will give it a shot (I wish this was suggested earlier). About the field: config_layout, gui_layout or gui_settings?

Reply:
> I wish this was suggested earlier

Yeah, my bad, I should have paid more attention to this.

> About the field: config_layout, gui_layout or gui_settings?

Maybe even something like `config_spec`.
I don't really have any strong opinions on the name, so just go with what sounds best to you.

Reply:
Ok, modified the settings plugin to support this and force pushed the new changes